### PR TITLE
KOA 5988 Migrate to gradle central dependency declaration

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: app/src/main/java/net/skyscanner/backpack/demo/meta/Component.kt [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 package net.skyscanner.backpack.demo.meta
 
-import javax.annotation.concurrent.Immutable
+import androidx.compose.runtime.Immutable
```
Comment (bvitaliyg) [inline app/src/main/java/net/skyscanner/backpack/demo/meta/Component.kt:21]: Imported wrong annotations previously

File: android-configuration.gradle [hunk 2]
```diff
@@ -48,8 +48,7 @@ android {
 }
 
 dependencies {
-    api rootProject.ext.androidxAnnotations
-    api "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```
Comment (marianeum) [inline android-configuration.gradle:52]: why was this removed?

File: app/build.gradle [hunk 3]
```diff
@@ -97,29 +97,35 @@ android {
 }
 
 dependencies {
+    def composeBom = platform(libs.compose.bom)
```
Comment (marianeum) [inline app/build.gradle:100]: this would've been better as a separate PR as it makes it hard/impossible to compare

File: app/build.gradle [hunk 4]
```diff
@@ -97,29 +97,35 @@ android {
...
+    implementation libs.square.picasso
+    implementation libs.accompanist.flowLayout
+    implementation libs.androidx.lifecycleKtx
+    implementation libs.compose.activity
+    implementation libs.kotlin.reflection
+    implementation libs.destinations.core
+    implementation libs.accompanist.pager
+    testImplementation libs.test.junit
+
+    debugImplementation composeBom
```
Comment (marianeum) [inline app/build.gradle:113]: do you actually have to re-declare it for everything? in the app we actually add the bom to all dependencies and declare it as an array instead, so you don't have to manually add it

File: backpack-lint/build.gradle [hunk 5]
```diff
@@ -20,12 +20,11 @@ apply plugin: 'kotlin'
 apply from: "tokens.gradle.kts"
 
 dependencies {
-    compileOnly "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```
Comment (marianeum) [inline backpack-lint/build.gradle:23]: again why was this removed?

File: gradle/libs.versions.toml
Comment (marianeum) [inline gradle/libs.versions.toml:1]: the dependabot config will need to be updated to make it work with this file

File: app/build.gradle (hunk 4 above)
Comment (bvitaliyg) [reply app/build.gradle:113]: It seems so:
https://discuss.gradle.org/t/applying-a-bom-platform-to-all-configurations/30621

You were able to do this because it was from the code. Toml just tells Gradle how to generate dependency providers and it cannot do this. Here's the example of the same from Compose website:
https://developer.android.com/jetpack/compose/bom/bom#does_the_bom_work_with_version_catalogs

File: app/build.gradle (hunk 3 above)
Comment (bvitaliyg) [reply app/build.gradle:100]: I can group the bomUsages in one block, so it's easier to review – WDYT?

File: gradle/libs.versions.toml
Comment (bvitaliyg) [reply gradle/libs.versions.toml:1]: No need to – it works with existing configs automatically – https://github.com/dependabot/dependabot-core/issues/3121#issuecomment-1467901381

However, I removed the block for buildSrc as it's not longer needed

**Review by marianeum — APPROVED** [review]
